modules/reactor/src/meshdivisions/HexagonalGridDivision.C

```diff
@@ -0,0 +1,194 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "HexagonalGridDivision.h"
+#include "MooseMesh.h"
+#include "HexagonalLatticeUtils.h"
+#include "Positions.h"
+
+#include "libmesh/elem.h"
+
+registerMooseObject("ReactorApp", HexagonalGridDivision);
+
+InputParameters
+HexagonalGridDivision::validParams()
+{
+  InputParameters params = MeshDivision::validParams();
+  params.addClassDescription(
+      "Divide the mesh along a hexagonal grid. Numbering of pin divisions increases first "
+      "counterclockwise, then expanding outwards from the inner ring, then axially. "
+      "Inner-numbering is within a radial ring, outer-numbering is axial divisions");
+
+  params.addParam<Point>("center", "Center of the hexagonal grid");
+  params.addParam<PositionsName>("center_positions", "Centers of the hexagonal grids");
+
+  params.addRequiredRangeCheckedParam<Real>(
+      "lattice_flat_to_flat",
+      "lattice_flat_to_flat>0",
+      "Distance between two (inner) opposite sides of a lattice. Also known as bundle pitch or "
+      "inner flat-to-flat distance");
+  params.addRequiredRangeCheckedParam<Real>("pin_pitch", "pin_pitch>0", "Distance between pins");
+
+  // Most of the infrastructure already exists to use another axis, it's just not disabled because
+  // not tested.
+  MooseEnum z_axis("X Y Z", "Z");
+  params.addParam<MooseEnum>("z_axis", z_axis, "Axial direction for the lattice");
+  params.suppressParameter<MooseEnum>("z_axis");
```

Comment:
I don't get this. why have this parameter?

Reply:
I can remove it

Reply:
but I started out wanting to support any axis. Then realized the utility is hardcoded to error on X and Y because it's not tested so surely not working